[ICU] build failure arm64-ios

**Host Environment**
- OS: OS X 12.5.1
- Compiler: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install --triplet=arm64-ios icu`

**Failure logs**

    Computing installation plan...
      The following packages will be built and installed:
          icu[core]:arm64-ios -> 71.1
      Detecting compiler hash for triplet arm64-ios...
      Restored 0 package(s) from /Users/kratky/Library/Caches/vcpkg/archives in 27.52 us. Use --debug to see more details.
      Installing 1/1 icu:arm64-ios...
      Building icu[core]:arm64-ios...
      warning: -- Using community triplet arm64-ios. This triplet configuration is not guaranteed to succeed.
      -- [COMMUNITY] Loading triplet configuration from: /Users/kratky/Documents/git/vcpkg/triplets/community/arm64-ios.cmake
      -- Using cached icu4c-71_1-src.tgz.
      -- Extracting source /Users/kratky/Library/Caches/vcpkg/downloads/icu4c-71_1-src.tgz
      -- Applying patch disable-escapestr-tool.patch
      -- Applying patch remove-MD-from-configure.patch
      -- Applying patch fix_parallel_build_on_windows.patch
      -- Applying patch fix-extra.patch
      -- Applying patch mingw-dll-install.patch
      -- Applying patch disable-static-prefix.patch
      -- Applying patch fix-win-build.patch
      -- Applying patch check-autoconf-archive.patch
      -- Using source at /Users/kratky/Documents/git/vcpkg/buildtrees/icu/src/c-71_1-src-08c83f7de1.clean
      -- Getting CMake variables for arm64-ios-dbg
      -- Getting CMake variables for arm64-ios-rel
      -- Generating configure for arm64-ios
      -- Finished generating configure for arm64-ios
      -- Configuring arm64-ios-dbg
      CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:96 (message):
          Command failed: /usr/local/bin/bash -c "V=1 ./../src/c-71_1-src-08c83f7de1.clean/source/configure  \"--disable-samples\" \"--disable-tests\" \"--disable-layoutex\" \"--with-cross-build=/Users/kratky/Documents/git/vcpkg/installed/x64-osx/tools/icu\" \"--disable-silent-rules\" \"--verbose\" \"--disable-shared\" \"--enable-static\" \"--enable-debug\" \"--disable-release\" \"--prefix=/Users/kratky/Documents/git/vcpkg/installed/arm64-ios/debug\" \"--bindir=\\${prefix}/../tools/icu/debug/bin\" \"--sbindir=\\${prefix}/../tools/icu/debug/sbin\" \"--libdir=\\${prefix}/lib\" \"--includedir=\\${prefix}/../include\" \"--datarootdir=\\${prefix}/share/icu\""
          Working Directory: /Users/kratky/Documents/git/vcpkg/buildtrees/icu/arm64-ios-dbg
          Error code: 77
          See logs for more information:
            /Users/kratky/Documents/git/vcpkg/buildtrees/icu/config-arm64-ios-dbg-config.log
            /Users/kratky/Documents/git/vcpkg/buildtrees/icu/config-arm64-ios-dbg-out.log
            /Users/kratky/Documents/git/vcpkg/buildtrees/icu/config-arm64-ios-dbg-err.log
  
      Call Stack (most recent call first):
        scripts/cmake/vcpkg_configure_make.cmake:797 (vcpkg_execute_required_process)
        ports/icu/portfile.cmake:53 (vcpkg_configure_make)
        scripts/ports.cmake:147 (include)
  
  
      error: building icu:arm64-ios failed with: BUILD_FAILED
      Please ensure you're using the latest port files with `git pull` and `vcpkg update`.
      Then check for known issues at:
          https://github.com/microsoft/vcpkg/issues?q=is%3Aissue+is%3Aopen+in%3Atitle+icu
      You can submit a new issue at:
          https://github.com/microsoft/vcpkg/issues/new?template=report-package-build-failure.md&title=[icu]+Build+error
      Include '[icu] Build error' in your bug report title, the following version information in your bug description, and attach any relevant failure logs from above.
          vcpkg-tool version: 2022-09-01-dfb82802c8cc562ce3b665a904a65b22314de724
          vcpkg-scripts version: a291bcad8 2022-09-07 (9 hours ago)



**Additional context**
Building other packages with `--triplet=arm64-ios` works without problems.

configure runs fine for me (version 72.1) but build fails. I assume this code should only be built for the build machine / tools.

```
[[maybe_unused]]
  ^
1 warning generated.
../.././../src/c-72_1-src-507f6288e7.clean/source/tools/pkgdata/pkgdata.cpp:553:18: error: 'system' is unavailable: not available on iOS
    int result = system(cmd);
                 ^
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS15.5.sdk/usr/include/stdlib.h:184:6: note: 'system' has been explicitly marked unavailable here
int      system(const char *) __DARWIN_ALIAS_C(system);
         ^
1 error generated.
make[2]: *** [pkgdata.o] Error 1
make[1]: *** [all-recursive] Error 2
make: *** [all-recursive] Error 2
```

--- a/ports/icu/portfile.cmake
+++ b/ports/icu/portfile.cmake
@@ -37,6 +37,11 @@ list(APPEND CONFIGURE_OPTIONS_DEBUG  --enable-debug --disable-release)
 set(RELEASE_TRIPLET ${TARGET_TRIPLET}-rel)
 set(DEBUG_TRIPLET ${TARGET_TRIPLET}-dbg)
 
+if("tools" IN_LIST FEATURES)
+  list(APPEND CONFIGURE_OPTIONS --enable-tools)
+else()
+  list(APPEND CONFIGURE_OPTIONS --disable-tools)
+endif()
 if(CMAKE_HOST_WIN32 AND VCPKG_TARGET_IS_MINGW AND NOT HOST_TRIPLET MATCHES "mingw")
     # Assuming no cross compiling because the host (windows) pkgdata tool doesn't
     # use the '/' path separator when creating compiler commands for mingw bash.

--- a/ports/icu/vcpkg.json
+++ b/ports/icu/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "icu",
   "version": "72.1",
+  "port-version": 1,
   "description": "Mature and widely used Unicode and localization library.",
   "homepage": "https://icu.unicode.org/home",
   "license": "ICU",
@@ -8,7 +9,15 @@
   "dependencies": [
     {
       "name": "icu",
-      "host": true
+      "host": true,
+      "features": [
+        "tools"
+      ]
     }
-  ]
+  ],
+  "features": {
+    "tools": {
+      "description": "Build tools"
+    }
+  }
 }

--- a/ports/qt5-base/vcpkg.json
+++ b/ports/qt5-base/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "qt5-base",
   "version": "5.15.8",
+  "port-version": 1,
   "description": "Qt5 Application Framework Base Module. Includes Core, GUI, Widgets, Networking, SQL, Concurrent and other essential qt components.",
   "homepage": "https://www.qt.io/",
   "license": null,
@@ -15,6 +16,9 @@
     "harfbuzz",
     {
       "name": "icu",
+      "features": [
+        "tools"
+      ],
       "platform": "!uwp"
     },
     "libjpeg-turbo",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -6206,7 +6206,7 @@
     },
     "qt5-base": {
       "baseline": "5.15.8",
-      "port-version": 0
+      "port-version": 1
     },
     "qt5-canvas3d": {
       "baseline": "0",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3022,7 +3022,7 @@
     },
     "icu": {
       "baseline": "72.1",
-      "port-version": 0
+      "port-version": 1
     },
     "ideviceinstaller": {
       "baseline": "1.1.2.23",

--- a/versions/i-/icu.json
+++ b/versions/i-/icu.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "2a4109603660116f3cb7477c234d3f0275124738",
+      "version": "72.1",
+      "port-version": 1
+    },
+    {
       "git-tree": "ca8792e25fbaf4f4d507eef05e2235b0449ed155",
       "version": "72.1",
       "port-version": 0

--- a/versions/q-/qt5-base.json
+++ b/versions/q-/qt5-base.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "3ca5efe99e5f582864670520199ec5da387ceb80",
+      "version": "5.15.8",
+      "port-version": 1
+    },
+    {
       "git-tree": "9bc09074d246485cea02dc5ba92ffcbae8f28cad",
       "version": "5.15.8",
       "port-version": 0

